src/components/pages/Home.tsx

```diff
@@ -1,13 +1,122 @@
-import React from 'react';
+import React, { useEffect, useRef } from 'react';
 import { Link } from 'react-router-dom';
 
 import '../styles/Home.scss';
 
+import * as THREE from 'three';
+import {OrbitControls} from 'three/examples/jsm/controls/OrbitControls';
+import {GLTFLoader} from 'three/examples/jsm/loaders/GLTFLoader';
+
+class Scene {
+  perspective: number;
+  container: HTMLCanvasElement;
+  scene: THREE.Scene;
+  renderer: THREE.WebGLRenderer;
+  camera!: THREE.PerspectiveCamera;
+  controls!: OrbitControls;
+
+  constructor() {
+    this.perspective = 800;
+    this.container = document.getElementById('stage') as HTMLCanvasElement;
+
+    this.scene = new THREE.Scene();
+    this.renderer = new THREE.WebGLRenderer({
+      canvas: this.container,
+      alpha: true,
+    });
+
+    this.renderer.setSize(window.innerWidth, window.innerHeight);
+    this.renderer.setPixelRatio(window.devicePixelRatio);
+
+    this.initLights();
+    this.initCamera();
+
+    this.controls = new OrbitControls(this.camera, this.container);
+
+    const loader = new GLTFLoader();
+    loader.load('/assets-3d/mug.glb', (gltf) => {
+      gltf.scene.position.set(0, -100, 0);
+      gltf.scene.children.forEach((mesh) => {
+        if (!mesh.material.name.includes('Buffer Layer material 1')) return;
+        mesh.material = new THREE.MeshPhongMaterial({
+          color: mesh.material.color,
+          shininess: 15,
+          reflectivity: .8,
+          refractionRatio: .5,
+          specular: 0x010101,
+        });
+      });
+      this.scene.add(gltf.scene);
+    });
+
+    window.addEventListener('resize', this.resize, false);
+    this.update();
+  }
+
+  private resize() {
+    this.camera.aspect = window.innerWidth / window.innerHeight;
```

Comment:
so we can't just replace this with a constant variable because we need to calculate the aspect ratio on the resize. 

but ill make a function that to return the aspect ratio

Reply:
Ohhh wait this maybe helpful for one of my problems!